fdbserver/SimulatedCluster.actor.cpp

```diff
@@ -645,6 +645,283 @@ T simulate(const T& in) {
 	return out;
 }
 
+ACTOR Future<std::string> getLayerStatus(Reference<ReadYourWritesTransaction> tr,
+                                         IPAddress localIP,
+                                         std::string name,
+                                         std::string id,
+                                         Database dest,
+                                         Snapshot snapshot = Snapshot::False) {
+	// This process will write a document that looks like this:
+	// { backup : { $expires : {<subdoc>}, version: <version from approximately 30 seconds from now> }
+	// so that the value under 'backup' will eventually expire to null and thus be ignored by
+	// readers of status.  This is because if all agents die then they can no longer clean up old
+	// status docs from other dead agents.
+
+	state Version readVer = wait(tr->getReadVersion());
+
+	state json_spirit::mValue layersRootValue; // Will contain stuff that goes into the doc at the layers status root
+	JSONDoc layersRoot(layersRootValue); // Convenient mutator / accessor for the layers root
+	JSONDoc op = layersRoot.subDoc(name); // Operator object for the $expires operation
+	// Create the $expires key which is where the rest of the status output will go
+
+	state JSONDoc layerRoot = op.subDoc("$expires");
+	// Set the version argument in the $expires operator object.
+	op.create("version") = readVer + 120 * CLIENT_KNOBS->CORE_VERSIONSPERSECOND;
```

Comment:
Says 30 seconds from now in comment above but here we have 120.  Should it be 30 or the comment needs update?